Patches/PlayerControlPatch.cs

```diff
@@ -118,6 +118,17 @@ public static bool Prefix(PlayerControl __instance, [HarmonyArgument(0)] PlayerC
             return false;
         }
 
+        Logger.Info($"Check: FirstDied", "ShieldPersonDiedFirst");
+
+        if (target.GetClient().GetHashedPuid() == Main.FirstDiedPrevious && MeetingStates.FirstMeeting)
+        {
+            killer.SetKillCooldown(5f);
+            killer.RpcGuardAndKill(target);
+            killer.Notify(Utils.ColorString(Utils.GetRoleColor(killer.GetCustomRole()), GetString("PlayerIsShieldedByGame")));
+            Logger.Info($"Canceled from ShieldPersonDiedFirst", "FirstDied");
+            return false;
+        }
```

Comment:
This code should be in `RpcCheckAndMurder` because CheckMurder only works on direct kills